src/components/FormElements/RevokeSponsorshipPicker.tsx

```diff
@@ -0,0 +1,294 @@
+import { Select } from "@stellar/design-system";
+
+import { Box } from "@/components/layout/Box";
+import { ExpandBox } from "@/components/ExpandBox";
+import { PubKeyPicker } from "@/components/FormElements/PubKeyPicker";
+import { AssetPicker } from "@/components/FormElements/AssetPicker";
+import { PositiveIntPicker } from "@/components/FormElements/PositiveIntPicker";
+import { TextPicker } from "@/components/FormElements/TextPicker";
+import { SignerPicker } from "@/components/FormElements/SignerPicker";
+
+import {
+  AnyObject,
+  RevokeSponsorshipValue,
+  SponsorshipType,
+} from "@/types/types";
+
+type RevokeSponsorshipPickerProps = {
+  id: string;
+  value: RevokeSponsorshipValue | undefined;
+  onChange: (value: RevokeSponsorshipValue | undefined) => void;
+  error: AnyObject | undefined;
+};
+
+export const RevokeSponsorshipPicker = ({
+  id,
+  value,
+  onChange,
+  error,
+}: RevokeSponsorshipPickerProps) => {
+  const fields: { id: SponsorshipType; label: string }[] = [
+    {
+      id: "account",
+      label: "Account",
+    },
+    {
+      id: "trustline",
+      label: "Trustline",
+    },
+    {
+      id: "offer",
+      label: "Offer",
+    },
+    {
+      id: "data",
+      label: "Data",
+    },
+    {
+      id: "claimable_balance",
+      label: "Claimable Balance",
+    },
+    {
+      id: "signer",
+      label: "Signer",
+    },
+  ];
+
+  const initData = (type: string) => {
+    switch (type) {
+      case "account":
+        return { account_id: undefined };
+      case "trustline":
+        return {
+          account_id: undefined,
+          asset: undefined,
+        };
+      case "offer":
+        return {
+          seller_id: undefined,
+          offer_id: undefined,
+        };
+      case "data":
+        return {
+          account_id: undefined,
+          data_name: undefined,
+        };
+      case "claimable_balance":
+        return { balance_id: undefined };
+      case "signer":
+        return {
+          account_id: undefined,
+          signer_key: undefined,
+        };
+      default:
+        return {};
+    }
+  };
+
+  const getInputId = (type: string, inputId: string) =>
+    `${id}-${type}-${inputId}`;
+
+  const renderInputs = () => {
+    if (!value?.type) {
+      return null;
+    }
+
+    switch (value.type) {
```

Comment:
I tried to make this piece more DRY, but it was harder to understand. So, I left it this way for clarity.